.pre-commit-config.yaml

```diff
@@ -0,0 +1,27 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+    # there are many log files in tests
+    # TODO: seperate py files and log files
+    -   id: trailing-whitespace
+        exclude: "^tests/.*$"
+    -   id: end-of-file-fixer
+        exclude: "^tests/.*$"
+    -   id: check-yaml
+        exclude: "^conda/.*$"
+    #-   id: check-json
```

Comment:
Why is `check-json` skipped?

Reply:
`check-json` does not allow duplicated keys. See https://github.com/pre-commit/pre-commit-hooks/blob/5191f112dfd133e97acba8c2162c37d8613cf54b/pre_commit_hooks/check_json.py#L15

Reply:
If we enable this hook, we need to rename _comment to _comment1, _comment2, ..., _commentN. We don't have the time to do so, but this line can be uncommented in a separated PR.